### PR TITLE
Campaign Overview Page Components

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

* Updates the auth flow to send authenticated users to `/campaigns` so they get taken directly to the overview page.
* Creates a `campaign_overview.blade.php` template to hold the overview page.
* Creates a `CampaignsController` to pass the overview data to the view
* Adds the overview component to `app.js` and adds it to the page when the `overviewContainer` is on the page
  - **questions** Is this the best way to do this? this will start a pattern of needing to look for a element on a page to know which page we are on and which components to use on it. 
* Creates a `CampaignTable` component that is basically this: 
image

* Uses hardcoded data past to the `window.STATE` object in the controller and then iterates over that data to show different tables. 

#### How should this be reviewed?

Pull down, build the frontend, log in, you should get to `/campaigns` it should look like this: 

image

#### Any background context you want to provide?

Full disclosure: I'm not sure we need to use react on this page. This will be a relatively static page that will be mostly used to navigate into the app. But it was a good exercise for me in using react. 

#### Relevant tickets
Addresses #151

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

### CODE REVIEW

File: resources/assets/app.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-  let message = `Hey ${props.message}`;
-  let subtitle = `${props.subtitle}`;
-
-  return <div className="container__block -narrow"><p>{ message }</p><h2>{subtitle}</h2></div>;
-};
+import CampaignOverview from './components/CampaignOverview'
 
 ready(() => {
-  ReactDom.render(<Greeter {...window.STATE} />, document.getElementById('app'));
+  const overviewContainer = document.getElementById('overviewContainer');
```
Comment (katiecrane) [inline resources/assets/app.js]: Ooooo I forget if I saw this in any of the tutorials I watched but this makes it so much easier to read when you call `render` later. I'm a big fan

File: resources/assets/app.js [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+
+  if(overviewContainer)
+  {
```
Comment (DFurnes) [inline resources/assets/app.js]: This is the most nitpicky of nitpicks, but our [style guide](https://github.com/airbnb/javascript#blocks--cuddled-elses) keeps braces on the same line for conditionals:

```js
if (overviewContainer) {
  ReactDom.render//...
}

File: resources/assets/components/CampaignOverview/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import CampaignTable from '../CampaignTable';
+
+class CampaignOverview extends React.Component {
+  render() {
+    const causeData = this.props;
+
+    const causeTables = Object.keys(causeData).map((cause, index) => {
```
Comment (DFurnes) [inline resources/assets/components/CampaignOverview/index.js]: This is some solid vanilla JS. 💪 We've also used Lodash's [`map`](https://lodash.com/docs/#map) helper for mapping over objects elsewhere, which is a little bit easier to read at first glance.

File: resources/assets/components/CampaignTable/Header.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+class Header extends React.Component {
+  render() {
+    return (
+      <thead>
+        <tr className="table__header">
```
Comment (DFurnes) [inline resources/assets/components/CampaignTable/Header.js]: Totally not necessary for this pull request, but it'd be lovely to make a general-purpose Table component for this that accepted an array of heading names & a row component and then formatted the data into it... something like this maybe:

```js
<Table headings={['Campaign Name', 'Approved', 'Pending', 'Rejected', 'Deleted']} row={Row} data={data} />
```

**Comment by DFurnes** [comment]
> **questions** Is this the best way to do this? this will start a pattern of needing to look for a element on a page to know which page we are on and which components to use on it.

Yeah, that'll be a good thing to keep an eye on. We had set up a little [`rehydrate` helper](https://github.com/DoSomething/voting-app/blob/dev/resources/assets/js/utilities/rehydrate.js) in the voting app that would look for any components on the page and render them if found, which was nice (especially since we did server pre-rendering on that app).

File: resources/assets/app.js (hunk 2 above)
Comment (sbsmith86) [inline resources/assets/app.js]: ack i always mess this up when switching between languages. Good 👁 . also looks like I need a 👾  there.

File: resources/assets/components/CampaignOverview/index.js (hunk 3 above)
Comment (sbsmith86) [inline resources/assets/components/CampaignOverview/index.js]: ah yes. LoDash.

File: resources/assets/components/CampaignTable/Header.js (hunk 4 above)
Comment (sbsmith86) [inline resources/assets/components/CampaignTable/Header.js]: oh coool. I definitely wanted a more general Table component but wasn't sure how to approach it. I will try this!

**Comment by sbsmith86** [comment]
@DFurnes Thanks for approving. I made some updates based on your comments about using Lodash and creating a general Table component. I am gonna merge this, but I will be continuing to work on this view so feel free to comment more if there is anything else you see.

**Pull request merged**